pkg/feature/raw_resources.go

```diff
@@ -16,56 +16,49 @@ package feature
 import (
 	"context"
 	"fmt"
-	"regexp"
 	"strings"
 
-	"github.com/ghodss/yaml"
 	"github.com/pkg/errors"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
 )
 
 const (
 	YamlSeparator = "(?m)^---[ \t]*$"
 )
 
-func (f *Feature) createResources(resources string) error {
-	splitter := regexp.MustCompile(YamlSeparator)
-	objectStrings := splitter.Split(resources, -1)
-	for _, str := range objectStrings {
-		if strings.TrimSpace(str) == "" {
-			continue
+func createResources(cli client.Client, objects []*unstructured.Unstructured, metaOptions ...cluster.MetaOptions) error {
+	for _, object := range objects {
+		for _, opt := range metaOptions {
+			if err := opt(object); err != nil {
+				return err // return immediately if any of the MetaOptions functions fail
+			}
 		}
-		u := &unstructured.Unstructured{}
-		if err := yaml.Unmarshal([]byte(str), u); err != nil {
-			return errors.WithStack(err)
-		}
-
-		ensureNamespaceIsSet(f, u)
 
-		name := u.GetName()
-		namespace := u.GetNamespace()
-
-		u.SetOwnerReferences([]metav1.OwnerReference{
-			f.AsOwnerReference(),
-		})
+		if !isNamespaceSet(object) {
+			return fmt.Errorf("no NS is set on %s", object.GetName())
+		}
```

Comment:
Please remind me: why are we changing the behavior here to fail when there's no ns set?

Reply:
I think this may be a mistake on my end, I will change it to set to `TargetNamespace` if not set

Reply:
This code was initially ported from `KfDef` implementation with the assumption this works as-is. There was a defaulting logic in case ns was not set.

I think we should just remove this check and rely on k8s error instead.

Reply:
Good thinking about cluster-scoped objects. I removed it entirely in 3c0a70c